google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -135,13 +184,42 @@ static com.google.monitoring.v3.TimeSeries convertTimeSeries(
         com.google.monitoring.v3.TimeSeries.newBuilder();
     builder.setResource(monitoredResourceBuilder.build());
     builder.setMetric(createMetric(metricName, metricTagKeys, metricTagValues));
-    builder.setMetricKind(createMetricKind(metricType));
-    builder.setValueType(createValueType(metricType));
-    Timestamp startTimeStamp = timeSeries.getStartTimestamp();
-    for (Point point : timeSeries.getPoints()) {
-      builder.addPoints(createPoint(point, startTimeStamp));
+
+    return builder;
+  }
+
+  private static com.google.monitoring.v3.TimeSeries.Builder setupBuilderForGceOrGKEResource(
+      MetricDescriptor metricDescriptor,
+      MonitoredResource gceOrGkeMonitoredResource,
+      TimeSeries timeSeries,
+      String clientId) {
+    Preconditions.checkNotNull(gceOrGkeMonitoredResource);
+    if (!Objects.equals(gceOrGkeMonitoredResource.getType(), GCE_RESOURCE_TYPE)
```

Comment:
is this possible? Should this check be moved to BigtableStackdriverStatsExporter#register? If the auto detected one is incorrect, can we populate it manually?